service/controller.go

```diff
@@ -231,6 +250,11 @@ func (s *service) CreateVolume(
 						continue
 					}
 
+					if sourceSystemID != "" && zoneTarget.systemID != sourceSystemID {
+						Log.Infof("systemID %s does not match snapshot/clone source systemID %s", zoneTarget.systemID, sourceSystemID)
+						continue
+					}
+
```

Comment:
does it make more sense to instead log success when we do find the matching systemID?
I think it might be confusing to see logs about a mismatch, but never see anything about a match.

That would increase complexity of the logic though...

Reply:
This was more for debugging, removing the log.

We already display a log which system will be used and the zone.